### DISCUSSION TITLE
Cant access flowise page on aws lightsail instance

### DISCUSSION THREAD

**Discussion by bruceb85** [FlowiseAI/Flowise#1700] [General]
Rather than using EC2 i am trying to use aws lightsail. 
What I've done:
I've installed and started flowise.
Set a static ip address for my instance. 
Open ports: 80, 443, 3000, 22

However, the address at port 3000 does not pull up any page. I've tried accessing it on both chrome and safari but nothing. 

I can run flowise on my local machine, but cant seem to run it on this ubuntu lightsail instance at aws. Does anyone have any recommendations?

**Comment by Jaredude** [comment]
I haven't done an AWS lightsail deployment (only EC2), but maybe try this with aws cli:

`aws lightsail open-instance-public-ports --instance-name YourInstanceName --port-info fromPort=3000,protocol=TCP,toPort=3000`

**Reply by bruceb85** [reply]
thanks for that, but it seems I have another issue now when trying to run flowise... it says this package is deprecated and flowise does not appear to start:

`ubuntu@ip-172-26-4-137:~$ npx flowise start --FLOWISE_USERNAME=admin --FLOWISE_PASSWORD=1234
 
Need to install the following packages:
flowise@1.5.0
Ok to proceed? (y) 
(⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂) ⠙ idealTree: timing arborist:ctor Completed in 1ms
npm WARN deprecated @oclif/screen@3.0.8: Package no longer supported. Contact Support at ht
tps://www.npmjs.com/support for more info.`

**Reply by Jaredude** [reply]
What version of node is it running? Are you able to install globally using yarn and run using yarn instead of using npx?

**Reply by bruceb85** [reply]
npm version: 8.5.1

[edit] node version: v12.22.9, yarn version: 1.22.21

I get the following when adding flowise with yarn:

`ubuntu@ip-172-26-10-254:~/.yarn/bin$ yarn add flowise
yarn add v1.22.21
info No lockfile found.
[1/4] Resolving packages...
warning flowise > @oclif/core > @oclif/screen@3.0.8: Package no longer supported. Contact S
upport at https://www.npmjs.com/support for more info.
warning flowise > flowise-components > puppeteer@20.9.0: < 21.5.0 is no longer supported
warning flowise > flowise-components > vm2@3.9.19: The library contains critical security i
ssues and should not be used for production! The maintenance of the project has been discon
tinued. Consider migrating your code to isolated-vm.
warning flowise > flowise-ui > react-code-blocks > react-syntax-highlighter > highlight.js@
9.15.10: Version no longer supported. Upgrade to @latest
warning flowise > flowise-ui > react-code-blocks > react-syntax-highlighter > lowlight > hi
ghlight.js@9.15.10: Version no longer supported. Upgrade to @latest
warning flowise > flowise-ui > flowise-embed-react > @ladle/react > @babel/plugin-proposal-
class-properties@7.18.6: This proposal has been merged to the ECMAScript standard and thus 
this plugin is no longer maintained. Please use @babel/plugin-transform-class-properties in
stead.
warning flowise > flowise-ui > rehype-mathjax > jsdom > abab@2.0.6: Use your platform's nat
ive atob() and btoa() methods instead
warning flowise > flowise-ui > rehype-mathjax > jsdom > domexception@4.0.0: Use your platfo
rm's native DOMException instead
warning flowise > flowise-ui > rehype-mathjax > jsdom > data-urls > abab@2.0.6: Use your pl
atform's native atob() and btoa() methods instead
warning flowise > flowise-components > llamaindex > rake-modified > fs-promise@2.0.3: Use m
z or fs-extra^3.0 with Promise Support
⠄ mri@^1.1.0Killed`


run flowise with yarn:
`ubuntu@ip-172-26-10-254:~/.yarn/bin$ yarn run flowise
yarn run v1.22.21
error Couldn't find a package.json file in "/home/ubuntu/.yarn/bin"`

**Reply by Jaredude** [reply]
npm is 8.5.1 but what version is node? node --version

If you do yarn add global flowise in the default dir of the server. Try the following:
cd
yarn add global flowise

If we get that far w/o any errors, then we're heading in the right direction and can try running flowise at that point. Otherwise, it could be that the version of Ubuntu lightsail needs more massaging to get it to work. Maybe a package registry or something.

**Reply by bruceb85** [reply]
I decided to go with ec2 instance as per the instructions, the dashboard now appears. However, one last thing I cant seem to figure out, is why I am not getting the username/password prompt on the web page.

I've added the following to the file: Flowise/docker/.env 
                                                                                             
```
PORT=3000
FLOWISE_USERNAME=user
FLOWISE_PASSWORD=1234
```

The docker-compose.yml file already has the following in it:

```
environment:
    - PORT=${PORT}
    - FLOWISE_USERNAME=${FLOWISE_USERNAME}
    - FLOWISE_PASSWORD=${FLOWISE_PASSWORD}
```

as per docker section: https://docs.flowiseai.com/configuration/authorization/app-level

**Reply by Jaredude** [reply]
And you did docker-compose down followed by docker-compose up -d?
